### PR TITLE
[Checkbox] indicator is inline element

### PR DESCRIPTION
**Pull request by giladgray**
- Checkbox alignment has always been a few pixels off from normal text, because the indicator relied heavily on line-height for centering (classic 2014).
- In this PR, control indicator is now an inline element and does not rely on line-height for positioning.
- Also refactor away from a bad `@extend` that generated very bloated CSS.

### before
height: 16px. padding/position.
image

### after
height: 18px (standard line-height). centered: perfect.
image

### CODE REVIEW

File: packages/core/src/components/forms/_controls.scss [hunk 1]
```diff
@@ -46,10 +93,10 @@ $control-checked-background-color-active: nth(map-get($button-intents, "primary"
   }
 
   .#{$ns}-control-indicator {
-    position: absolute;
-    top: 0;
-    left: 0;
-    margin: 0;
+    display: inline-block;
+    position: relative;
+    margin-top: -3px;
```
Comment (themadcreator) [inline packages/core/src/components/forms/_controls.scss]: does this depend of font size?

File: packages/core/src/components/forms/_controls.scss [hunk 2]
```diff
@@ -46,10 +93,10 @@ $control-checked-background-color-active: nth(map-get($button-intents, "primary"
   }
 
   .#{$ns}-control-indicator {
-    position: absolute;
-    top: 0;
-    left: 0;
-    margin: 0;
+    display: inline-block;
```
Comment (themadcreator) [inline packages/core/src/components/forms/_controls.scss]: this seems pretty major

Comment (giladgray) [reply packages/core/src/components/forms/_controls.scss]: it is, but it's also wholly contained inside the component. no API has been changed.

File: packages/core/src/components/forms/_controls.scss (hunk 1 above)
Comment (giladgray) [reply packages/core/src/components/forms/_controls.scss]: 😕 it does, but checkboxes were not designed to support arbitrary font size. that would require a different approach. i've been fiddling with a flex layout a la menu items (requiring wrapping children in a `<span>`) but i don't think it's worth the change just yet.

**Pull request merged**